documentation/api.tex

```diff
@@ -0,0 +1,69 @@
+\chapter{API documentation}
+This chapter will cover the documentation of the back-end API.
+
+\section{GraphiQL}
+When using \gls{graphql}, you get \gls{graphiql} out of the box.
+This means that you can have a live documentation of your schema that you can use both to browse the documentation but also query against the application.
+This is very useful for developers. Both for the back-end developer that does not need to maintain a document with his API documentation.
+But also for the front-end developer that can browse the schema live instead of constantly needing to ask for a new copy of the API documentation.
+
+On figure \ref{fig:graphiql} an example of how you can use \gls{graphiql} is shown. To the far right the documentation explorer is viewed.
+The center column is where the \gls{graphql} responses are shown.
+The column to the far left is where you can write your queries.
+In this example, 2 queries are already defined and named. With this you can use the \verb+Play+ button to run one of them seperately.
+The data in the middle column is from the query login. This shows that it only returns exactly the 2 parameters i asked for when i queried.
+
+\graphic{1}{api/graphiql}{The GraphiQL interface in a browser}{fig:graphiql}
+
+\section{Root}
+On figure \ref{fig:root} the two root types are shown. 
+These are Query and Mutation. 
+Query can be considered as GET requests and Mutations can be considered as POST requests.
+\graphich{0.6}{api/rootType}{Root types}{fig:root}
+
+\section{Query}
+On figure \ref{fig:query} the \verb+Query+ root type is shown. It contains two fields. 
+
+The first field \verb+viewer+ is where the restricted data is located.
+It takes a token string, and returns a viewer type which is shown on figure \ref{fig:viewerType}.
+
+The second field \verb+login+ also returns a viewer type. 
+This is to save a round trip.
+The viewer type has a \verb+token+ field that can be used to query the \verb+viewer+ field at a later time after logging in.
+\graphich{0.6}{api/queryType}{The Query root type with fields}{fig:query}
+
+\section{Viewer}
+The \verb+Viewer+ can only be retrieved if the user is authenticated. 
+Either through the \verb+viewer+ field or \verb+login+ field on the root type \verb+query+. The \verb+Viewer+ is shown on figure \ref{fig:viewerType}
+\graphich{0.6}{api/viewerType}{The Query root type with fields}{fig:viewerType}
+
+\section{Mutation}
+The \verb+Mutation+ root type is also a restricted field where the user needs to be authenticated. 
+It takes a \verb+token+ string which is required. 
+Required is shown by the ! after String. 
```

Comment:
GraphQL notation. Either move this up to the introduction to the API documentation or leave it out.